webapps/frontend/ui/admin/client/scripts/pages/execution-metrics.js

```diff
@@ -17,190 +17,358 @@
 
 'use strict';
 
-var template = require('./execution-metrics.html?raw');
-var CamSDK = require('camunda-bpm-sdk-js/lib/angularjs/index');
+const template = require('./execution-metrics.html?raw');
+const moment = require('moment');
+const angular = require('angular');
 
-var debouncePromiseFactory = require('camunda-bpm-sdk-js').utils
+const debouncePromiseFactory = require('camunda-bpm-sdk-js').utils
   .debouncePromiseFactory;
-var debounceQuery = debouncePromiseFactory();
+const debounceMonthly = debouncePromiseFactory();
+const debounceAnnual = debouncePromiseFactory();
 
-var Controller = [
+const fmtMonth = 'MMMM YYYY';
+const fmtYear = 'DD/MM/YYYY';
+const fmtDatePicker = 'yyyy-MM-dd';
+const fmtRequest = 'YYYY-MM-DD';
+const metrics = {
+  PI: 'process-instances',
+  DI: 'decision-instances',
+  TU: 'task-users',
+  FNI: 'flow-node-instances',
+  EDE: 'executed-decision-elements'
+};
+const localConfContractStartDate = 'metricsContractStartDate';
+
+const Controller = [
   '$scope',
   '$filter',
-  'Uri',
   'camAPI',
-  'fixDate',
   'configuration',
-  function($scope, $filter, Uri, camAPI, fixDate, configuration) {
-    var MetricsResource = camAPI.resource('metrics');
-
-    // date variables
-    var now = new Date();
-    var dateFilter = $filter('date');
-    var dateFormat = "yyyy-MM-dd'T'HH:mm:ss";
+  'localConf',
+  'PluginMetricsResource',
+  '$translate',
+  function(
+    $scope,
+    $filter,
+    camAPI,
+    configuration,
+    localConf,
+    PluginMetricsResource,
+    $translate
+  ) {
+    const telemetryResource = camAPI.resource('telemetry');
+    const dateFilter = $filter('date');
 
     // initial scope data
-    $scope.startDate = dateFilter(
-      now.getFullYear() + '-01-01T00:00:00.000',
-      dateFormat
-    );
-    $scope.endDate = dateFilter(
-      now.getFullYear() + '-12-31T23:59:59.999',
-      dateFormat
-    );
+    $scope.fmtDatePicker = fmtDatePicker;
+    let startDate = localConf.get(localConfContractStartDate);
+    if (startDate) {
+      $scope.startDate = startDate;
+    } else {
+      startDate = moment()
+        .startOf('year')
+        .toDate();
+    }
+    $scope.startDate = dateFilter(startDate, fmtDatePicker);
+
     $scope.loadingState = 'INITIAL';
```

Comment:
🔧  I think `$scope.loadingState` might not be used anymore and could be removed

Reply:
Removed.